src/MissingTableHelper.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Keboola\GenericExtractor;
+
+use Keboola\Csv\CsvFile;
+use Keboola\GenericExtractor\Exception\UserException;
+
+class MissingTableHelper
+{
+    public static function checkConfigs($configs, $dataDir)
+    {
+        foreach ($configs as $config) {
+            if ($config->getAttribute('mappings')) {
+                foreach ($config->getAttribute('mappings') as $name => $mapping) {
+                    $destinationBase = $dataDir . '/out/tables/' . $config->getAttribute('outputBucket') . '.';
+                    self::fillMissingTableMapping(
+                        $destinationBase,
+                        $config->getAttribute('outputBucket'),
+                        $config->getAttribute('incremental'),
+                        $name,
+                        $mapping
+                    );
+                }
+            }
+        }
+    }
+
+    private static function fillMissingTableMapping(
+        $baseFileName,
+        $outputBucket,
+        $incremental,
+        $name,
+        $mapping,
+        $parentKey = []
+    ) {
+        $columns = [];
+        $primaryKey = [];
+        foreach ($mapping as $itemName => $item) {
+            if (empty($item['type'])) {
+                $item['type'] = 'column';
+            }
+            if ($item['type'] === 'table') {
+                self::fillMissingTableMapping(
+                    $baseFileName,
+                    $outputBucket,
+                    $incremental,
+                    $item['destination'],
+                    $item['tableMapping'],
+                    empty($item['parentKey']) ? ['destination' => $name . '_pk'] : $item['parentKey']
+                );
+            } elseif (($item['type'] === 'column') || ($item['type'] === 'user')) {
```

Comment:
Small comment.  I think you can move the empty check to this condition -> `empty($item['type']) ||  === $item['type'] === 'column' || ....` if you put it first.